tests/phpunit/tests/query/postStatus.php

```diff
@@ -457,4 +459,162 @@ public function test_specific_post_should_be_returned_if_trash_is_one_of_the_req
 
 		$this->assertContains( $p1, wp_list_pluck( $q->posts, 'ID' ) );
 	}
+
+	/**
+	 * @ticket 48556
+	 * @ticket 13509
+	 */
+	public function test_non_singular_queries_using_post_type_any_should_respect_post_type_read_private_posts_cap() {
```

Comment:
The posts can be set up as shared fixtures in `wpSetUpBeforeClass()` to save a bit of time.

You'll need to register the post types in both the before class and setUp function due to an issue with how the test suite resets globals.

Reply:
I avoided that because I didnt want to mess with all the other tests. What do you think of having a `setUpMultipleTests` method only for these 2 tests?

Reply:
I pushed up to use shared fixtures in 4d676b54a922c8341577acf3ae0381a7b1997103 -- I added a few lines to get all posts available to avoid messing up other tests.

Creating the various objects in `wpSetUpBeforeClass` means they only need to be created once, so the common function for these tests would still lose any performance gain from using shared fixtures.

There's a lot going on here so I am still testing a few more things.